src/model_repository_manager/model_lifecycle.cc

```diff
@@ -559,11 +560,21 @@ ModelLifeCycle::CreateModel(
   // backend.
   if (!model_config.backend().empty()) {
     std::unique_ptr<TritonModel> model;
+    const uint64_t model_load_ns =
+        std::chrono::duration_cast<std::chrono::nanoseconds>(
+            std::chrono::steady_clock::now().time_since_epoch())
+            .count();
+    model_info->load_start_ns_ = model_load_ns;
     status = TritonModel::Create(
         server_, model_info->model_path_, options_.backend_cmdline_config_map,
         options_.host_policy_map, model_id, version, model_config,
         is_config_provided, &model);
     is.reset(model.release());
+    if (status.IsOk()) {
+#ifdef TRITON_ENABLE_METRICS
+      CalculateAndReportLoadTime(model_info, &is);
+#endif  // TRITON_ENABLE_METRICS
+    }
```

Comment:
If the `load_start_ns` is only for reporting load time, how about we simply pass the `load_start_ns` directly into `CalculateAndReportLoadTime()` function? i.e.
```c++
CalculateAndReportLoadTime(model_load_ns, &is);
```
without storing it in `model_info->load_start_ns_` first, because it will not be accessed again after reporting the load time.

Reply:
Yes makes sense now as we pass model pointer for metrics reporter sperately.